### PR TITLE
Codechange: Replace duplicated code with TileArea::Expand()

### PR DESCRIPTION
**Pull request by PeterN**
Code for expanding an OrthogonalTileArea is duplicated throughout the code, sometimes missing bounds checks. This PR adds a function to expand with bounds checking, and uses it where possible.

### CODE REVIEW

File: src/tilearea.cpp [hunk 1]
```diff
@@ -117,6 +117,25 @@ bool OrthogonalTileArea::Contains(TileIndex tile) const
 	return IsInsideBS(tile_x, left, this->w) && IsInsideBS(tile_y, top, this->h);
 }
 
+/**
+ * Expand a tile area by rad tiles in each direction, keeping within map bounds.
+ * @param rad Number of tiles to expand
+ */
+void OrthogonalTileArea::Expand(int rad)
```
Comment (LordAro) [inline src/tilearea.cpp]: might simplify some code if this returns `OrthogonalTileArea &`/`this` ?

Comment (PeterN) [reply src/tilearea.cpp]: Yea. I'll do that.
Although that implies it doesn't modify `this`?

Comment (LordAro) [reply src/tilearea.cpp]: not really? enables method chaining, is all

Comment (PeterN) [reply src/tilearea.cpp]: Hmm, this doesn't seem to help any of the changes in this PR.

I think I'd have to change calls from `TileArea ta(tile, w, h); ta.Expand(r);` to `TileArea ta = TileArea(tile, w, h).Expand(r);` Not sure this is better.

**Comment by nielsmh** [comment]
Some of the Kdtree queries should probably use this too.

File: src/tilearea.cpp (hunk 1 above)
Comment (LordAro) [reply src/tilearea.cpp]: could use `auto ta = ...` to remove the duplicated type. Up to you, but I think if you don't use method chaining, there's no point giving Expand a return type

Comment (PeterN) [reply src/tilearea.cpp]: I kinda dislike using auto, except for complex template-iterator types. So...

**Review by michicc — APPROVED** [review]

**Pull request merged**